src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -6598,6 +6598,13 @@ public void ClearContent(string path)
                 throw PSTraceSource.NewArgumentException("path");
             }
 
+            if (Directory.Exists(path))
+            {
+                string errorMsg = StringUtil.Format(SessionStateStrings.ClearDirectoryContent, path);
+                WriteError(new ErrorRecord(new NotSupportedException(errorMsg), "ClearDirectoryContent",ErrorCategory.InvalidOperation, path));
+                return;
+            }
+
             path = NormalizePath(path);
```

Comment:
Should the check happen after normalizing the path?

Reply:
I think that will be safe, will change it.